### PR TITLE
Refactor to shared projects and add new VSIX project for Visual Studio 2022

### PR DESCRIPTION
**Pull request by BertanAygun**
Ports the extension to Visual Studio 2022. Due to breaking changes, VSIX projects now utilize shared projects and there are 2 VSIX generating projects to produce output for VS 2019 and earlier and VS 2022 and above.

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -4,5 +4,6 @@
     <clear />
     <add key="dotnet-public" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-public/nuget/v3/index.json" />
     <add key="myget-legacy" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/myget-legacy/nuget/v3/index.json" />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
```
Comment (BertanAygun) [inline NuGet.Config:7]: This had to be added because many of the newer packages referenced by VS 2022 SDK wasn't found in dotnet-public feed.

**Comment by brumlemann** [comment]
Would be nice to have access to this in VS2022. Any idea on the time schedule for this PR to be merged?

**Comment by bartsipes** [comment]
Seems like this would fix #985 which a lot of people are looking for (as well as #1001 and #1006).

**Comment by terrajobst** [comment]
Closing as API Port was deprecated in favor of binary analysis in [.NET Upgrade Assistant](https://github.com/dotnet/upgrade-assistant/blob/main/docs/binary_analysis.md).

**Pull request closed without merging**